### ISSUE TITLE
Potential Bug with shouldDirtyConsider and a Disabled Field

### ISSUE THREAD

**Issue by ericsakmar** [edmundhung/conform#703]
### Describe the bug and the expected behavior

Hello! It seems that `shouldDirtyConsider` is not being correctly applied when a field has been disabled.

I've pasted a Stackblitz below to show the problem. `form.dirty` is immediately `true` even though nothing else has changed and the `shouldDirtyConsider` is set to return `true` for any field other than `email`.

Thanks for reading!

### Conform version

v1.1.5

### Steps to Reproduce the Bug or Issue

[Stackblitz](https://stackblitz.com/edit/remix-run-remix-gel12n?file=app%2Froutes%2F_index.tsx)

**Comment by edmundhung** [comment]
Thanks for the PR, @ericsakmar! I am happy with the changes you made. But I would like to learn more about your usecase first before I have it merged.

The issue you had with `shouldDirtyConsider` was caused by the missing entry when the input is disabled. Although your PR does fix the dirty state, the validation is still failing due to missing value from the form data. I wonder if you are looking for a way to stop user from editing the input value, maybe you can use the `readonly` attribute instead?

**Comment by ericsakmar** [comment]
Yes, the `readonly` attribute does solve the problem I thought I had. Thanks!
